WordPress/gutenberg
# Scripts: Update `lighthouse` dependency

## What?

Updates the `lighthouse` dependency in the `e2e-test-utils-playwright` package.

## Why?


This fixes the issue with high-severity vulnerabilities introduced with an older version of the `puppeteer-core` dependency, as mentioned in #64597. Lighthouse uses puppeteer-core and needs to be updated to use the latest version to remove the old dependencies with vulnerabilities.

## How?


- Updates the e2e-test-utils-playwright's `package.json` lighthouse dependency to the latest version.
- Removes the `lighthouse+10.4.0.patch` file the old version depended on.

## Testing Instructions





Same as #64597:

1.  Run `npm i --save-dev @wordpress/scripts`
2.  Run `npm audit` and you'll see a warning for high-severity vulnerabilities.
3.  Add the following to your package.json file:
```json
"overrides": {
    "puppeteer-core": "^23.1.0"
}
```
4. Run `npm i` to update the packages.
5. The vulnerabilities are now removed.

## Other Notes

1. I'm not sure this package can be updated with a simple change in this PR. Someone with more familiarity can chime in.  Issue #60357 mentions puppeteer and its removal, so if that's in progress and includes the `e2e-test-utils-playwright` package, we can disregard this PR.

2. The core package.json and package-lock.json may need to be updated to include lighthouse as a dev dependency as was the case for #64597.
